core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -105,6 +105,17 @@ package object config {
     .bytesConf(ByteUnit.MiB)
     .createOptional
 
+  private[spark] val DRIVER_MEMORY_OVERHEAD_FACTOR =
+    ConfigBuilder("spark.driver.memoryOverheadFactor")
+      .doc("The amount of non-heap memory to be allocated per driver in cluster mode, " +
+      "as a fraction of total driver memory. If memory overhead is specified directly, " +
+      "it takes precedence.")
+      .version("3.3.0")
+      .doubleConf
+      .checkValue(factor => factor >= 0 && factor < 1,
+        "Ensure that memory overhead is a double between 0 --> 1.0")
```

Comment:
`factor > 0` would should be sufficient, we can have cases where overhead is higher than jvm Xmx.